add ellipticity definition explanation in the FoF matching tutorial

yymao opened the pull request:
This PR fixes #76. See the new plot in #76. 

Probably need to add a reference on where the HSM ellipticity definition comes from (added).

rmandelb commented on tutorials/matching_fof.ipynb:
```diff
@@ -424,6 +424,41 @@
+    "# compare the ellipticity (smartly)\n",
+    "conversion = lambda e: 2*e / (1.0+e*e)\n",
+    "plt.figure(figsize=(5,5));\n",
+    "plt.scatter(conversion(matched_gals['ellipticity_true']), matched_gals['shape_hsm_regauss_etot'], s=1);\n",
```
This is a super minor point (and isn't about a new aspect of the notebook), but I thought that `ellipticity_true` is before the application of lensing shear, while the observed shape is after, so it's not totally apples-to-apples for that reason as well?  Perhaps this is worth noting in the discussion of the shapes in the catalog, so people realize that this distinction could be relevant for some purposes, unless another tutorial already covers the distinction between unlensed and lensed quantities in the extragalactic catalogs.

yymao commented:
@rmandelb Thanks for the comments. I've addressed all of them in the latest commit and this PR is ready for review again.

It turns out that lensed ellipticity is not directly available in the extragalactic catalog. Considering that this is a tutorial, I added some more cells to describe the trick of adding user-define quantities using GCR. It's small diversion but I hope that's ok.

yymao commented on tutorials/matching_fof.ipynb:
```diff
@@ -424,6 +424,126 @@
+   "outputs": [],
+   "source": [
+    "def calc_lensed_ellipticity(e1, e2, shear1, shear2):\n",
+    "    return np.hypot(e1+shear1, e2+shear2) \n",
```
Thanks, @rmandelb. I've updated `calc_lensed_ellipticity` in commit 918d01d to address this issue. I no longer use the small-shear approximate but use the full calculation (which I hope I implemented correctly). 

(I didn't opt to use GalSim because it doesn't do array operations.)

yymao commented:
Ready for re-review.
